eng/native/configuretools.cmake

```diff
@@ -108,3 +108,12 @@ if (CLR_CMAKE_HOST_LINUX)
     set(CLR_CMAKE_TARGET_LINUX_MUSL 1)
   endif()
 endif()
+
+# Common configuration for all runtimes, set runtime feature flag
+string(TOLOWER "${PROJECT_NAME}" PROJECT_NAME_LOWER)
+
+if(PROJECT_NAME_LOWER STREQUAL "coreclr")
+    set(CLR_CMAKE_RUNTIME_CORECLR 1)
```

Comment:
@jkoritzinsky Is this going to break invariants we want to maintain for VMR?

Reply:
I believe this will work today, but I would like to remove it long term.

Given @filipnavara's comment at https://github.com/dotnet/runtime/pull/110471/files/a5d1115e4eb9f3b7dfe3e1534c97ed647bd68bf6#r1920334843, the one use case might be easily fixed, so we could remove it then.

Reply:
I applied my fixed on top of this branch and removed the `CLR_CMAKE_RUNTIME_CORECLR` usage. It is indeed trivial change. Note that https://github.com/dotnet/runtime/pull/109928 is ready to be merged. It will likely create some tiny conflicts with this PR.

We still need to take care of `CLR_CMAKE_RUNTIME_MONO` though.